paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/activity/PaymentSheetPlaygroundActivity.kt

```diff
@@ -0,0 +1,204 @@
+            application
+        )
+    }
+
+    private val customer: Repository.CheckoutCustomer
+        get() = when (viewBinding.customerRadioGroup.checkedRadioButtonId) {
+            R.id.guest_customer_button -> Repository.CheckoutCustomer.New
+            R.id.new_customer_button -> {
+                viewModel.temporaryCustomerId?.let {
+                    Repository.CheckoutCustomer.WithId(it)
+                } ?: Repository.CheckoutCustomer.New
+            }
+            else -> Repository.CheckoutCustomer.Returning
+        }
+
+    private val googlePayConfig: PaymentSheet.GooglePayConfiguration?
+        get() = when (viewBinding.googlePayRadioGroup.checkedRadioButtonId) {
+            R.id.google_pay_on_button -> {
+                PaymentSheet.GooglePayConfiguration(
+                    environment = PaymentSheet.GooglePayConfiguration.Environment.Test,
+                    countryCode = "US"
+                )
+            }
+            else -> null
+        }
+
+    private val mode: Repository.CheckoutMode
+        get() = when (viewBinding.modeRadioGroup.checkedRadioButtonId) {
+            R.id.mode_payment_button -> Repository.CheckoutMode.Payment
+            else -> Repository.CheckoutMode.Setup
+        }
+
+    private val merchantName = "Example, Inc."
+
+    private lateinit var paymentSheet: PaymentSheet
+    private lateinit var flowController: PaymentSheet.FlowController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(viewBinding.root)
+
+        paymentSheet = PaymentSheet(this, ::onPaymentSheetResult)
+        flowController = PaymentSheet.FlowController.create(
+            this,
+            ::onPaymentOption,
+            ::onPaymentSheetResult
+        )
+
+        viewBinding.reloadButton.setOnClickListener {
+            lifecycleScope.launch {
+                viewModel.prepareCheckout(customer, mode)
+            }
+        }
+
+        viewBinding.completeCheckoutButton.setOnClickListener {
+            startCompleteCheckout()
+        }
+
+        viewBinding.customCheckoutButton.setOnClickListener {
+            flowController.confirm()
+        }
+
+        viewModel.status.observe(this) {
+            Toast.makeText(this, it, Toast.LENGTH_LONG).show()
+        }
+
+        viewModel.inProgress.observe(this) {
+            viewBinding.progressBar.isInvisible = !it
+        }
+
+        viewModel.readyToCheckout.observe(this) { isReady ->
+            if (isReady) {
+                viewBinding.completeCheckoutButton.isEnabled = true
+                configureCustomCheckout()
+            } else {
+                disableViews()
+            }
+        }
+    }
+
+    private fun disableViews() {
+        viewBinding.completeCheckoutButton.isEnabled = false
+        viewBinding.customCheckoutButton.isEnabled = false
+        viewBinding.paymentMethod.setOnClickListener(null)
```

Comment:
Seems we never sets the clickListener back, I think disabling the button will block onClick events, is there any reason we nullify it?